Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsStore.cpp

```diff
@@ -1228,6 +1249,104 @@ bool ResourceLoadStatisticsStore::insertObservedDomain(const ResourceLoadStatist
     return true;
 }
 
+void ResourceLoadStatisticsStore::insertLoginStatus(RegistrableDomain&& domain, IsLoggedIn loggedInStatus, std::optional<LoginStatus>&& lastAuthentication)
+{
+    ASSERT(!RunLoop::isMain());
+
+    if (loginStatusDomainID(domain)) {
+        ITP_RELEASE_LOG_ERROR("insertLoginStatus: failed to find domain");
+        return;
+    }
+
+    enum {
+        DomainIDIndex,
+        RegistrableDomainIndex,
+        UserNameIndex,
+        TokenTypeIndex,
+        AuthenticationTypeIndex,
+        LoggedInTimeIndex,
+        TimeToLiveIndex,
+        StatusIndex
+    };
+
+    auto scopedStatement = this->scopedStatement(m_insertLoginstatusStatement, insertLoginStatusQuery, "insertLoginstatus"_s);
+    if (!scopedStatement) {
+        ITP_RELEASE_LOG_DATABASE_ERROR("insertLoginStatus: failed to create scoped statement");
+        return;
+    }
+
+    if (scopedStatement->bindText(RegistrableDomainIndex, domain.string()) != SQLITE_OK) {
+        ITP_RELEASE_LOG_DATABASE_ERROR("insertLoginStatus: failed to bind registrableDomain");
+        return;
+    }
+
+    if (lastAuthentication) {
+        if (scopedStatement->bindText(UserNameIndex, lastAuthentication->username()) != SQLITE_OK
+            || scopedStatement->bindInt(TokenTypeIndex, static_cast<int>(lastAuthentication->tokenType())) != SQLITE_OK
+            || scopedStatement->bindInt(AuthenticationTypeIndex, static_cast<int>(lastAuthentication->authType())) != SQLITE_OK
+            || scopedStatement->bindDouble(LoggedInTimeIndex, lastAuthentication->loggedInTime().secondsSinceEpoch().value()) != SQLITE_OK
+            || scopedStatement->bindDouble(TimeToLiveIndex, lastAuthentication->timeToLive().seconds()) != SQLITE_OK) {
+            ITP_RELEASE_LOG_DATABASE_ERROR("insertLoginStatus: failed to bind lastAuthentication values");
+            return;
+        }
+    } else {
+        if (scopedStatement->bindNull(UserNameIndex) != SQLITE_OK
+            || scopedStatement->bindNull(TokenTypeIndex) != SQLITE_OK
+            || scopedStatement->bindNull(AuthenticationTypeIndex) != SQLITE_OK
+            || scopedStatement->bindNull(LoggedInTimeIndex) != SQLITE_OK
+            || scopedStatement->bindNull(TimeToLiveIndex) != SQLITE_OK) {
+            ITP_RELEASE_LOG_DATABASE_ERROR("insertLoginStatus: failed to bind NULL values for lastAuthentication");
+            return;
+        }
+    }
+
+    if (scopedStatement->bindInt(StatusIndex, static_cast<int>(loggedInStatus)) != SQLITE_OK) {
+        ITP_RELEASE_LOG_DATABASE_ERROR("insertLoginStatus: failed to bind status");
+        return;
+    }
+
+    if (scopedStatement->step() != SQLITE_DONE) {
+        ITP_RELEASE_LOG_DATABASE_ERROR("insertLoginStatus: failed to step statement");
+        return;
+    }
+    return;
+}
+
+void ResourceLoadStatisticsStore::removeLoginStatus(RegistrableDomain&& domain)
```

Comment:
Seems all these functions do not need to take a `RegistrableDomain&&`, a `const RegistrableDomain&` would suffice. Also, `WTFMove(domain)` is pointless  to call `loginStatusDomainID()` since it takes in a `const RegistrableDomain&`.